### ISSUE TITLE
Grammar gets distracted by custom environments

### ISSUE THREAD

**Issue by PhilippImhof** [ashthespy/Atom-LaTeX#251]
Consider the following minimal document:

```latex
\documentclass{article}

\usepackage{tikz}
\newenvironment{foo}{\begin{tikzpicture}}{\end{tikzpicture}}

\begin{document}

\begin{foo}
  \draw(0,0) circle (1cm);
\end{foo}

\end{document}
```

This compiles without any errors. However, with the current grammar, every line after `\newenvironment` is in scope `meta.function.environment.latex.tikz` and `meta.group.braces.tex` which is not correct.

**Comment by ashthespy** [comment]
The current syntax highlighting is accomplished [using a set of regex rules](https://github.com/ashthespy/Atom-LaTeX/blob/851698065a3cf4caaadc394202a46a4ec5db743e/grammars/latex.cson#L469-L489) and doesn't really parse the TeX. 
It thus never finds the second argument of  `/end{tikzpicture}` to `newenvironment`.  It might be possible to add a rule for `newenvironment`, but to be honest IIRC these grammars were just converted from the TextMate bundle, so I am not sure..

**Comment by PhilippImhof** [comment]
Yes, I saw that. It might however be worth making a change, e.g. reacting on `\newenvironment` and thus not taking into account the `\begin{tikzpicture}`, because it does not open an environment.
